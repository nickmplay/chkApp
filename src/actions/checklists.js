//checklist action generators

//add newChecklist object for dispatch
export const addChecklist = (newChecklist) => ({
  type: 'ADD_CHECKLIST',
  newChecklist
});

//add removeChecklist object for dispatch
export const removeChecklist = (checklistToRemoveId) => ({
  type: 'REMOVE_CHECKLIST',
  checklistToRemoveId
});