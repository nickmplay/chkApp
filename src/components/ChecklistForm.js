import React from 'react';

export class ChecklistForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      command: props.command,
      name: props.checklist ? props.checklist.name : '',
      items: props.checklist ? props.checklist.items.join(',') : ''
    };
  }
  onNameChange = (e) => {
    const name = e.target.value;
    this.setState(() => ({ name }));
  };
  onItemChange = (e) => {
    const newItem = e.target.value;
    this.setState(() => ({ items: newItem }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    //form validation

    this.props.onSubmit({
      name: this.state.name,
      items: this.state.items.split(',')
    });
  };

  render() {
    return (
      <div className="content-container">
        <form className="form" onSubmit={this.onSubmit}>
          <input
            type='text'
            placeholder='Checklist Name'
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <textarea
            type='text'
            placeholder='Add Items, seperating each with a comma. Eg Item 1, Item 2, Item 3'
            value={this.state.items}
            onChange={this.onItemChange}
          />
          <button className="button">{this.state.command} Checklist</button>
        </form>
      </div>
    );
  }
}