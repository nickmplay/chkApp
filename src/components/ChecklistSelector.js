import React from 'react';
import { Link } from 'react-router-dom';

export const ChecklistSelector = (props) => {
  return (
    <div>
      {props.checklists.map(checklist => {
        return (
          <div className="content-container list-item" key={checklist.id}>
            <Link to={`/view/${checklist.id}`}>
              <p  key={checklist.id + 1}>{checklist.name}</p>
            </Link>
            <Link key={checklist.id + 2} to={`/edit/${checklist.id}`}>Edit</Link>
          </div>
        );
      })}
    </div>);
};