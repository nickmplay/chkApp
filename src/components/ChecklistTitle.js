import React from 'react';
import { countChecklists } from '../selectors/checklist';

export const ChecklistTitle = (props) => {
  const checklistsWord = props.checklists.length === 1 ? 'checklist' : 'checklists';

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">You have <span>{props.checklists.length}</span> {checklistsWord} to use:</h1>
        {/* <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div> */}
      </div>
    </div>
  );
};

export default ChecklistTitle;
