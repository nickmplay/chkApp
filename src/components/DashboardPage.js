import React from 'react';

const DashboardPage = () => (
  <div className="content-container">
    <h1>You have two checklists to use:</h1>
    <div>Checklist 1</div>
    <div>Checklist 2</div>
  </div>
);

export default DashboardPage;
