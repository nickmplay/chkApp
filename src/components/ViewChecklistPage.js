import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';


export class ViewChecklistPage extends React.Component {
  onToggle = (e) => {
    if (e.target.className == 'chk-unselected') {
      e.target.className = 'chk-selected';
    } else {
      e.target.className = 'chk-unselected';
    }
  }

  render() {
    return (
      <div className="content-container">
        <h1>Viewing {this.props.checklist[0].name}</h1>
        {this.props.checklist[0].items.map((e, i) => <p className='chk-unselected' onClick={this.onToggle} key={i}> {e}</p>)}
        <Link to="/">Return to dashboard</Link>
      </div>
    )
  };
};

const mapStateToProps = (state, props) => ({
  checklist: state.checklists.filter(e => e.id.toString() === props.match.params.id)
});

export default connect(mapStateToProps)(ViewChecklistPage);
