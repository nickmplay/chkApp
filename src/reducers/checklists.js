//checklist reducer function

const defaultChecklistState = [];

export default (state = defaultChecklistState, action) => {
  switch (action.type) {
    case 'ADD_CHECKLIST':
      return [...state, action.newChecklist];
    case 'REMOVE_CHECKLIST':
      return state.filter(({ id }) => id !== action.checklistToRemoveId);
    case 'EDIT_CHECKLIST':
      return state.map((e) => {
        if (e.id === action.checklistUpdates.id) {
          return {
            id: e.id,
            items: action.checklistUpdates.items
          };
        } else {
          return e; 
        }
      });
    default:
      return state;
  }
};
