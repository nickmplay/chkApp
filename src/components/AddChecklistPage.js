import React from 'react';
import { connect } from 'react-redux';
import { addChecklist } from '../actions/checklists';
import { startAddChecklist } from '../actions/firebase';
import { ChecklistForm } from './ChecklistForm';

export class AddChecklistPage extends React.Component {
  onSubmit = ({name, items}) => {
    this.props.startAddChecklist({
      name,
      items
    });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Checklist</h1>
          </div>
        </div>
        <ChecklistForm 
          command="Add"
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddChecklist: (checklist) => dispatch(startAddChecklist(checklist))
});

export default connect(undefined, mapDispatchToProps)(AddChecklistPage);
