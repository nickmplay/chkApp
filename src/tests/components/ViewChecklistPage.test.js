import React from 'react';
import { shallow } from 'enzyme';
import { ViewChecklistPage } from '../../components/ViewChecklistPage';
import checklists from '../fixtures/checklists';

test('should render ViewChecklistPage correctly with data', () => {
  const wrapper = shallow(<ViewChecklistPage checklist={[checklists[0]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should change class upon clicking', () => {
  const onToggle = jest.fn();
  const targetMock = {
    target: { className: 'chk-unselected' }
  };
  const wrapper = shallow(<ViewChecklistPage
    onToggle={onToggle}
    checklist={[checklists[0]]}
  />);

  expect(wrapper).toMatchSnapshot();
  wrapper.find('p').at(0).simulate('click', targetMock);
  expect(targetMock).toEqual({
    target: { className: 'chk-selected' }
  });
  // expect(onToggle).toHaveBeenCalled();

  wrapper.find('p').at(0).simulate('click', targetMock);
  expect(targetMock).toEqual({
    target: { className: 'chk-unselected' }
  });
  // expect(onToggle).toHaveBeenCalled();
});