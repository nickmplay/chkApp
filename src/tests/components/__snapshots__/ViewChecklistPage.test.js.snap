// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change class upon clicking 1`] = `
<div
  className="content-container"
>
  <h1>
    Viewing 
    bed time
  </h1>
  <p
    className="chk-unselected"
    key="0"
    onClick={[Function]}
  >
     
    lock back door
  </p>
  <p
    className="chk-unselected"
    key="1"
    onClick={[Function]}
  >
     
    lock front door
  </p>
  <p
    className="chk-unselected"
    key="2"
    onClick={[Function]}
  >
     
    turn off lights
  </p>
  <Link
    replace={false}
    to="/"
  >
    Return to dashboard
  </Link>
</div>
`;

exports[`should render ViewChecklistPage correctly with data 1`] = `
<div
  className="content-container"
>
  <h1>
    Viewing 
    bed time
  </h1>
  <p
    className="chk-unselected"
    key="0"
    onClick={[Function]}
  >
     
    lock back door
  </p>
  <p
    className="chk-unselected"
    key="1"
    onClick={[Function]}
  >
     
    lock front door
  </p>
  <p
    className="chk-unselected"
    key="2"
    onClick={[Function]}
  >
     
    turn off lights
  </p>
  <Link
    replace={false}
    to="/"
  >
    Return to dashboard
  </Link>
</div>
`;
